### ISSUE TITLE
Current x86_64 binary needs both libpng12 and libpng16

### ISSUE THREAD

**Issue by akien-mga** [TurningWheel/Barony#27]
See:
```
$ ldd barony.x86_64 
        linux-vdso.so.1 (0x00007ffee279b000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f653b5f3000)
        libSDL2-2.0.so.0 => /media/data/Jeux/SteamApps/common/Barony/libSDL2-2.0.so.0 (0x00007f653b2b9000)
        libSDL2_image-2.0.so.0 => /media/data/Jeux/SteamApps/common/Barony/libSDL2_image-2.0.so.0 (0x00007f653b093000)
        libSDL2_net-2.0.so.0 => /media/data/Jeux/SteamApps/common/Barony/libSDL2_net-2.0.so.0 (0x00007f653ae8e000)
        libSDL2_ttf-2.0.so.0 => /media/data/Jeux/SteamApps/common/Barony/libSDL2_ttf-2.0.so.0 (0x00007f653ac86000)
        libsteam_api.so => /media/data/Jeux/SteamApps/common/Barony/libsteam_api.so (0x00007f653aa62000)
        libGLU.so.1 => /lib64/libGLU.so.1 (0x00007f653a7e3000)
        libGL.so.1 => /lib64/libGL.so.1 (0x00007f653a578000)
        libpng12.so.0 => /lib64/libpng12.so.0 (0x00007f653a34f000)
        libfmodex.so => /media/data/Jeux/SteamApps/common/Barony/libfmodex.so (0x00007f6539f9f000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f6539c9f000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f6539a88000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f65396d5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f653b970000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f65394d1000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f65392b4000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f65390ac000)
        libfreetype.so.6 => /lib64/libfreetype.so.6 (0x00007f6538e13000)
        libexpat.so.1 => /lib64/libexpat.so.1 (0x00007f6538be9000)
        libxcb-dri3.so.0 => /lib64/libxcb-dri3.so.0 (0x00007f65389e6000)
        libxcb-present.so.0 => /lib64/libxcb-present.so.0 (0x00007f65387e3000)
        libxcb-sync.so.1 => /lib64/libxcb-sync.so.1 (0x00007f65385dc000)
        libxshmfence.so.1 => /lib64/libxshmfence.so.1 (0x00007f65383d9000)
        libglapi.so.0 => /lib64/libglapi.so.0 (0x00007f65381aa000)
        libXext.so.6 => /lib64/libXext.so.6 (0x00007f6537f98000)
        libXdamage.so.1 => /lib64/libXdamage.so.1 (0x00007f6537d95000)
        libXfixes.so.3 => /lib64/libXfixes.so.3 (0x00007f6537b8f000)
        libX11-xcb.so.1 => /lib64/libX11-xcb.so.1 (0x00007f653798d000)
        libX11.so.6 => /lib64/libX11.so.6 (0x00007f6537650000)
        libxcb.so.1 => /lib64/libxcb.so.1 (0x00007f653742a000)
        libxcb-glx.so.0 => /lib64/libxcb-glx.so.0 (0x00007f6537211000)
        libxcb-dri2.so.0 => /lib64/libxcb-dri2.so.0 (0x00007f653700c000)
        libXxf86vm.so.1 => /lib64/libXxf86vm.so.1 (0x00007f6536e06000)
        libdrm.so.2 => /lib64/libdrm.so.2 (0x00007f6536bf7000)
        libz.so.1 => /media/data/Jeux/SteamApps/common/Barony/libz.so.1 (0x00007f65369e1000)
        libbz2.so.1 => /lib64/libbz2.so.1 (0x00007f65367d1000)
        libpng16.so.16 => /media/data/Jeux/SteamApps/common/Barony/libpng16.so.16 (0x00007f653659b000)
        libXau.so.6 => /lib64/libXau.so.6 (0x00007f6536397000)
        libXdmcp.so.6 => /lib64/libXdmcp.so.6 (0x00007f6536191000)
```

libpng16 should be sufficient. I think the barony binary itself was compiled against libpng12, and libpng16 is pulled in by system dependencies like freetype, since:
```
$ ldd *.so* | grep png
libpng16.so.16:
        libpng16.so.16 => /lib64/libpng16.so.16 (0x00007fd15d9c8000)
```
no bundled library seems to require it apart from itself :)

**Comment by addictgamer** [comment]
That's interesting.

On my system, it seems to just build against libpng16. Don't know why it was pulling in libpng12 for the release build.

**Comment by akien-mga** [comment]
Fixed in the current 2.0.4 build on Steam:
```
[akien@cauldron Barony]$ ldd barony.x86_64 | grep png
        libpng16.so.16 => /media/data/Jeux/SteamApps/common/Barony/libpng16.so.16 (0x00007f3be1177000)
[akien@cauldron Barony]$ ldd editor.x86_64 | grep png
        libpng16.so.16 => /media/data/Jeux/SteamApps/common/Barony/libpng16.so.16 (0x00007f48d6ae5000)
```

**Comment by akien-mga** [comment]
Congrats on the release btw :)
